nuxeo-mongodb-directory/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -167,10 +255,32 @@ public void deleteEntry(DocumentModel docModel) throws DirectoryException {
 
     @Override
     public void deleteEntry(String id) throws DirectoryException {
+        deleteEntry(directoryName, id);
+    }
+
+    @Override
+    public void deleteEntry(String id, Map<String, String> map) throws DirectoryException {
+        deleteEntry(id);
+    }
+
+    public void deleteEntry(String collection, String id) throws DirectoryException {
         checkPermission(SecurityConstants.WRITE);
         checkDeleteConstraints(id);
+
+        for (Reference reference : getDirectory().getReferences()) {
+            if (reference instanceof MongoDBReference) {
+                MongoDBReference mongoDBReference = (MongoDBReference) reference;
+                mongoDBReference.removeLinksForSource(id, this);
+            } else {
+                reference.removeLinksForSource(id);
+            }
+        }
+
         try {
-            DeleteResult result = getCollection().deleteOne(
+            if (StringUtils.isBlank(collection)) {
```

Comment:
At first, I was thinking of exposing this method so that it could be used elsewhere with a different collection than the directory name (in my case, in the reference). But it was in fact not necessary, we can call direclty the delete method from the mongo client. Thus, this check becomes also not necessary.